commands/admin.js

```diff
@@ -0,0 +1,137 @@
+                .setDescription("WARNING: Removes all course roles from the server."),
+        ),
+    async execute(interaction) {
+        try {
+            if (!interaction.member.permissions.has(Permissions.FLAGS.ADMINISTRATOR)) {
+                return await interaction.reply({
+                    content: "You do not have permission to execute this command.",
+                    ephemeral: true,
+                });
+            }
+
+            if (interaction.options.getSubcommand() === COMMAND_KICKUNVERIFIED) {
+                const role = await interaction.guild.roles.cache.find(
+                    (r) => r.name.toLowerCase() === "unverified",
+                );
+
+                // Make sure that the "unverified" role exists
+                if (role === undefined) {
+                    return await interaction.reply('Error: no "unverified" role exists.');
+                }
+
+                const kickMessage =
+                    "You have been removed from the CSESoc Server as you have not verified via the instructions in #welcome.\
+                    If you wish to rejoin, visit https://cseso.cc/discord";
+
+                // Member list in the role is cached
+                let numRemoved = 0;
+                await role.members.each((member) => {
+                    member.createDM().then((DMChannel) => {
+                        // Send direct message to user being kicked
+                        DMChannel.send(kickMessage).then(() => {
+                            // Message sent, time to kick.
+                            member
+                                .kick(kickMessage)
+                                .then(() => {
+                                    ++numRemoved;
+                                    console.log(numRemoved + " people removed.");
+                                })
+                                .catch((e) => {
+                                    console.log(e);
+                                });
+                        });
+                    });
+                });
+                return await interaction.reply("Removed unverified members.");
+            } else if (interaction.options.getSubcommand() === COMMAND_MIGRATE) {
+                // get all roles, and find courses which match the regex
+                const course_roles = await interaction.guild.roles.cache.filter((role) =>
+                    is_valid_course(role.name),
+                );
+
+                // for each course role, find the corresponding channel
+                // and add a permission overwrite for each member with that role
+                // to that channel so that they can view it
+
+                for (const role of course_roles.values()) {
+                    const channel = await interaction.guild.channels.cache.find(
+                        (role_channel) =>
+                            role_channel.name.toLowerCase() === role.name.toLowerCase(),
```

Comment:
This would rely on the fact that role names match up exactly to course channel names. From what I can see course channel names have been changed to reflect that e.g. comp-seng4920 -> comp4920, but just wanted to put it out there if there any channels that I can't see that don't follow these conventions.

Reply:
yes, I thought of this and fixed all such instances